// ng4-material-dropdown/dist/src/modules/components/dropdown/ng4-dropdown.ngfactory.js
"use strict";
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("ng4-material-dropdown/dist/src/modules/services/dropdown-state.service");
var i2 = require("ng4-material-dropdown/dist/src/modules/components/dropdown/ng4-dropdown");
var styles_Ng4Dropdown = [];
exports.RenderType_Ng4Dropdown = i0.ɵcrt({ encapsulation: 2, styles: styles_Ng4Dropdown,
    data: {} });
function View_Ng4Dropdown_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 5, 'div', [['class',
                'ng4-dropdown-container']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n    '])), i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n    '])), i0.ɵncd(null, 1), (_l()(),
            i0.ɵted(null, ['\n'])), (_l()(), i0.ɵted(null, ['\n']))], null, null);
}
exports.View_Ng4Dropdown_0 = View_Ng4Dropdown_0;
function View_Ng4Dropdown_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'ng4-dropdown', [], null, [['window', 'scroll']], function (_v, en, $event) {
            var ad = true;
            if (('window:scroll' === en)) {
                var pd_0 = (i0.ɵnov(_v, 2).scrollListener() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, View_Ng4Dropdown_0, exports.RenderType_Ng4Dropdown)), i0.ɵprd(512, null, i1.DropdownStateService, i1.DropdownStateService, []), i0.ɵdid(114688, null, 2, i2.Ng4Dropdown, [i1.DropdownStateService], null, null), i0.ɵqud(335544320, 1, { button: 0 }),
        i0.ɵqud(335544320, 2, { menu: 0 })], function (_ck, _v) {
        _ck(_v, 2, 0);
    }, null);
}
exports.View_Ng4Dropdown_Host_0 = View_Ng4Dropdown_Host_0;
exports.Ng4DropdownNgFactory = i0.ɵccf('ng4-dropdown', i2.Ng4Dropdown, View_Ng4Dropdown_Host_0, { dynamicUpdate: 'dynamicUpdate' }, { onItemClicked: 'onItemClicked',
    onItemSelected: 'onItemSelected', onShow: 'onShow', onHide: 'onHide' }, ['ng4-dropdown-button',
    'ng4-dropdown-menu']);
