// ng4-material-dropdown/dist/src/modules/components/button/ng4-dropdown-button.ngfactory.js
"use strict";
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("./style.scss.shim.ngstyle");
var i1 = require("@angular/core");
var i2 = require("ng4-material-dropdown/dist/src/modules/components/button/ng4-dropdown-button");
var i3 = require("@angular/common");
var styles_Ng4DropdownButton = [i0.styles];
exports.RenderType_Ng4DropdownButton = i1.ɵcrt({ encapsulation: 0,
    styles: styles_Ng4DropdownButton, data: {} });
function View_Ng4DropdownButton_1(_l) {
    return i1.ɵvid(0, [(_l()(), i1.ɵeld(0, null, null, 10, 'span', [['class',
                'ng4-dropdown-button__caret']], null, null, null, null, null)), (_l()(), i1.ɵted(null, ['\n        '])), (_l()(), i1.ɵeld(0, null, null, 7, ':svg:svg', [[':xml:space', 'preserve'], [':xmlns:xlink',
                'http://www.w3.org/1999/xlink'], ['enable-background', 'new 0 0 32 32'], ['height',
                '16px'], ['id', 'Слой_1'], ['version', '1.1'], ['viewBox', '0 0 32 32'], ['width',
                '16px'], ['xmlns', 'http://www.w3.org/2000/svg']], null, null, null, null, null)), (_l()(), i1.ɵeld(0, null, null, 0, ':svg:path', [['d', 'M24.285,11.284L16,19.571l-8.285-8.288c-0.395-0.395-1.034-0.395-1.429,0  c-0.394,0.395-0.394,1.035,0,1.43l8.999,9.002l0,0l0,0c0.394,0.395,1.034,0.395,1.428,0l8.999-9.002  c0.394-0.395,0.394-1.036,0-1.431C25.319,10.889,24.679,10.889,24.285,11.284z'],
            ['fill', '#121313'], ['id', 'Expand_More']], null, null, null, null, null)), (_l()(), i1.ɵeld(0, null, null, 0, ':svg:g', [], null, null, null, null, null)), (_l()(), i1.ɵeld(0, null, null, 0, ':svg:g', [], null, null, null, null, null)), (_l()(),
            i1.ɵeld(0, null, null, 0, ':svg:g', [], null, null, null, null, null)), (_l()(), i1.ɵeld(0, null, null, 0, ':svg:g', [], null, null, null, null, null)), (_l()(), i1.ɵeld(0, null, null, 0, ':svg:g', [], null, null, null, null, null)), (_l()(), i1.ɵeld(0, null, null, 0, ':svg:g', [], null, null, null, null, null)), (_l()(), i1.ɵted(null, ['\n    ']))], null, null);
}
function View_Ng4DropdownButton_0(_l) {
    return i1.ɵvid(0, [(_l()(), i1.ɵeld(0, null, null, 9, 'button', [['class',
                'ng4-dropdown-button'], ['tabindex', '0s'], ['type', 'button']], null, [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.toggleMenu() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i1.ɵted(null, ['\n    '])), (_l()(),
            i1.ɵeld(0, null, null, 3, 'span', [['class', 'ng4-dropdown-button__label']], null, null, null, null, null)),
        (_l()(), i1.ɵted(null, ['\n        '])), i1.ɵncd(null, 0), (_l()(),
            i1.ɵted(null, ['\n    '])), (_l()(), i1.ɵted(null, ['\n\n    '])),
        (_l()(), i1.ɵand(16777216, null, null, 1, null, View_Ng4DropdownButton_1)),
        i1.ɵdid(16384, null, 0, i3.NgIf, [i1.ViewContainerRef, i1.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i1.ɵted(null, ['\n'])), (_l()(), i1.ɵted(null, ['\n']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.showCaret;
        _ck(_v, 8, 0, currVal_0);
    }, null);
}
exports.View_Ng4DropdownButton_0 = View_Ng4DropdownButton_0;
function View_Ng4DropdownButton_Host_0(_l) {
    return i1.ɵvid(0, [(_l()(), i1.ɵeld(0, null, null, 1, 'ng4-dropdown-button', [], null, null, null, View_Ng4DropdownButton_0, exports.RenderType_Ng4DropdownButton)), i1.ɵdid(49152, null, 0, i2.Ng4DropdownButton, [i1.ElementRef], null, null)], null, null);
}
exports.View_Ng4DropdownButton_Host_0 = View_Ng4DropdownButton_Host_0;
exports.Ng4DropdownButtonNgFactory = i1.ɵccf('ng4-dropdown-button', i2.Ng4DropdownButton, View_Ng4DropdownButton_Host_0, { showCaret: 'showCaret' }, { onMenuToggled: 'onMenuToggled' }, ['*']);
