// ng4-material-dropdown/dist/src/modules/components/menu/ng4-dropdown-menu.ngfactory.js
"use strict";
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("./style.scss.shim.ngstyle");
var i1 = require("@angular/core");
var i2 = require("@angular/common");
var i3 = require("ng4-material-dropdown/dist/src/modules/components/menu/ng4-dropdown-menu");
var i4 = require("ng4-material-dropdown/dist/src/modules/services/dropdown-state.service");
var styles_Ng4DropdownMenu = [i0.styles];
exports.RenderType_Ng4DropdownMenu = i1.ɵcrt({ encapsulation: 0,
    styles: styles_Ng4DropdownMenu, data: { 'animation': [{ type: 7, name: 'fade', definitions: [{ type: 0,
                        name: 'visible', styles: { type: 6, styles: { display: 'block', overflow: '*' }, offset: null },
                        options: undefined }, { type: 0, name: 'hidden', styles: { type: 6, styles: { display: 'none',
                                overflow: 'hidden', width: '0' }, offset: null }, options: undefined },
                    { type: 1, expr: 'hidden => visible', animation: [{ type: 4, styles: { type: 5, steps: [{ type: 6,
                                            styles: { opacity: 0, offset: 0, height: '0', width: '0' }, offset: null },
                                        { type: 6, styles: { opacity: 1, offset: 1, height: '*', width: '*' }, offset: null }] },
                                timings: 150 }], options: null }, { type: 1, expr: 'visible => hidden',
                        animation: [{ type: 4, styles: { type: 5, steps: [{ type: 6, styles: { opacity: 1, offset: 0,
                                                height: '*', width: '*' }, offset: null }, { type: 6, styles: { opacity: 0,
                                                offset: 1, height: '0', width: '0' }, offset: null }] }, timings: 250 }],
                        options: null }], options: {} }, { type: 7, name: 'opacity', definitions: [{ type: 1,
                        expr: 'hidden => visible', animation: [{ type: 4, styles: { type: 5, steps: [{ type: 6,
                                            styles: { opacity: 0, offset: 0 }, offset: null }, { type: 6, styles: { opacity: 1,
                                                offset: 1 }, offset: null }] }, timings: 450 }], options: null },
                    { type: 1, expr: 'visible => hidden', animation: [{ type: 4, styles: { type: 5, steps: [{ type: 6,
                                            styles: { opacity: 1, offset: 0 }, offset: null }, { type: 6, styles: { opacity: 0.5,
                                                offset: 0.3 }, offset: null }, { type: 6, styles: { opacity: 0, offset: 1 },
                                            offset: null }] }, timings: 200 }], options: null }], options: {} }] } });
function View_Ng4DropdownMenu_1(_l) {
    return i1.ɵvid(0, [(_l()(), i1.ɵeld(0, null, null, 0, 'div', [['class',
                'ng4-dropdown-backdrop']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.hide() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null))], null, null);
}
function View_Ng4DropdownMenu_0(_l) {
    return i1.ɵvid(0, [(_l()(), i1.ɵted(null, ['\n'])), (_l()(), i1.ɵeld(0, null, null, 6, 'div', [], [[8, 'className', 0], [2, 'ng4-dropdown-menu--inside-element',
                null], [2, 'ng4-dropdown-menu--open', null], [24, '@fade', 0]], null, null, null, null)), (_l()(), i1.ɵted(null, ['\n        '])), (_l()(), i1.ɵeld(0, null, null, 3, 'div', [['class',
                'ng4-dropdown-menu__options-container']], [[24, '@opacity', 0]], null, null, null, null)), (_l()(), i1.ɵted(null, ['\n            '])),
        i1.ɵncd(null, 0), (_l()(), i1.ɵted(null, ['\n        '])), (_l()(),
            i1.ɵted(null, ['\n'])), (_l()(), i1.ɵted(null, ['\n\n'])),
        (_l()(), i1.ɵted(null, ['\n'])), (_l()(), i1.ɵand(16777216, null, null, 1, null, View_Ng4DropdownMenu_1)), i1.ɵdid(16384, null, 0, i2.NgIf, [i1.ViewContainerRef, i1.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i1.ɵted(null, ['\n']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_5 = _co.state.menuState.isVisible;
        _ck(_v, 11, 0, currVal_5);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = i1.ɵinlineInterpolate(1, 'ng4-dropdown-menu ng4-dropdown-menu---width--', _co.width, '');
        var currVal_1 = !_co.appendToBody;
        var currVal_2 = _co.state.menuState.isVisible;
        var currVal_3 = _co.state.menuState.toString();
        _ck(_v, 1, 0, currVal_0, currVal_1, currVal_2, currVal_3);
        var currVal_4 = _co.state.menuState.toString();
        _ck(_v, 3, 0, currVal_4);
    });
}
exports.View_Ng4DropdownMenu_0 = View_Ng4DropdownMenu_0;
function View_Ng4DropdownMenu_Host_0(_l) {
    return i1.ɵvid(0, [(_l()(), i1.ɵeld(0, null, null, 2, 'ng4-dropdown-menu', [], null, null, null, View_Ng4DropdownMenu_0, exports.RenderType_Ng4DropdownMenu)), i1.ɵdid(507904, null, 1, i3.Ng4DropdownMenu, [i4.DropdownStateService, i1.ElementRef, i1.Renderer], null, null),
        i1.ɵqud(603979776, 1, { items: 1 })], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, null);
}
exports.View_Ng4DropdownMenu_Host_0 = View_Ng4DropdownMenu_Host_0;
exports.Ng4DropdownMenuNgFactory = i1.ɵccf('ng4-dropdown-menu', i3.Ng4DropdownMenu, View_Ng4DropdownMenu_Host_0, { width: 'width', focusFirstElement: 'focusFirstElement',
    offset: 'offset', appendToBody: 'appendToBody' }, {}, ['*']);
